server/app/auth/ApiAuthenticator.java

```diff
@@ -100,11 +100,13 @@ public void validate(Credentials rawCredentials, WebContext context, SessionStor
       throwUnauthorized(context, "API key is expired: " + keyId);
     }
 
-    if (!isAllowedIp(apiKey, resolveClientIp(context))) {
+    String resolvedIp = resolveClientIp(context);
+    if (!isAllowedIp(apiKey, resolvedIp)) {
       throwUnauthorized(
           context,
           String.format(
-              "IP %s not in allowed range for key ID: %s", context.getRemoteAddr(), keyId));
+              "Resolved IP %s is not in allowed range for key ID: %s, which is \"%s\"",
```

Comment:
I added the allowed subnet range to the log for debugging convenience, but lmk if you think it shouldn't be included.

Reply:
Fine to include it. If an attacker has access to server logs there are bigger problems